dap-python.el

```diff
@@ -70,6 +70,140 @@ as the pyenv version then also return nil. This works around https://github.com/
           executable))
     (executable-find command)))
 
+
+
+(cl-defstruct point
+  (line (:type integer) :named t)
+  (character (:type integer) :named t))
+
+
+(cl-defstruct location
+  (start (:type point) :named t)
+  (end (:type point) :named t))
+
+
+(cl-defstruct lsp-symbol
+  (name (:type string) :named t)
+  (type (:type string) :named t)
+  (location (:type location) :named t))
+
+
+(cl-defgeneric == (lhs rhs)
+  (:documentation "Check if lhs and rhs are equal"))
+
+
+(cl-defmethod == ((lhs symbol) (rhs symbol))
+  (eq lhs rhs))
+
+
+(cl-defmethod == ((lhs integer) (rhs integer))
+  (eq lhs rhs))
+  
+
+(cl-defmethod == ((lhs string) (rhs string))
+  (string-equal lhs rhs))
+
+
+(cl-defmethod == ((lhs list) (rhs list))
+  (and (== (length lhs) (length rhs))
+       (-reduce (lambda (x y) (and x y)) (-zip-with '== lhs rhs))))
+
+
+(cl-defmethod == ((lhs point) (rhs point))
+  (and (== (point-line lhs) (point-line rhs))
+       (== (point-character lhs) (point-character rhs))))
+
+
+(cl-defmethod == ((lhs location) (rhs location))
+  (and (== (location-start lhs) (location-start rhs))
+       (== (location-end lhs) (location-end rhs))))
+
+
+(cl-defmethod == ((lhs lsp-symbol) (rhs lsp-symbol))
+  (and (== (lsp-symbol-name lhs) (lsp-symbol-name rhs))
+       (== (lsp-symbol-type lhs) (lsp-symbol-type rhs))
+       (== (lsp-symbol-location lhs) (lsp-symbol-location rhs))))
+
+
+
+(defun dap-python--parse-lsp-symbol (symbol)
+  (-let* (((&hash "name" "kind" "location") symbol)
```

Comment:
IMO we should do this either automatically or not at all. As a side note, we could use dash to add custom destructuring, for example and validate the stuff compile time with no runtime penalty. 

``` emacs-lisp
(defun dash-expand:&lsp-point (key source)
  (unless (member key '(:line :character))
    (error "Unknown key:  %s" key))
  `(gethash ,(substring (symbol-name key) 1) ,source))

;; works 
(-let [(&lsp-point :line) (ht ("line" 10)
                              ("character" 10))]
  line)

;; fails compile time
(-let [(&lsp-point :line2) (ht ("line" 10)
                               ("character" 10))]
  line2)
```